src/TransactionData.java

```diff
@@ -0,0 +1,83 @@
+public class TransactionData {
+
+    private final Object[] transactionData = new Object[] {
+            new Object[] { TransactionType.CREDIT, "Dress Refund", 2999 },
+            new Object[] { TransactionType.CREDIT, "Pocket Money", 250 },
+            new Object[] { TransactionType.CREDIT, "Lottery winnings", 5600000 },
+            new Object[] { TransactionType.CREDIT, "TV Appearance", 100000 },
+            new Object[] { TransactionType.CREDIT, "Book Deal", 500000 },
+            new Object[] { TransactionType.CREDIT, "Poker winnings", 80000 },
+            new Object[] { TransactionType.CREDIT, "Paper Round", 2300 },
+            new Object[] { TransactionType.CREDIT, "Let flat", 420000 },
+            new Object[] { TransactionType.DEBIT, "Lottery Ticket", -300 },
+            new Object[] { TransactionType.DEBIT, "Train Ticket", -12330 },
+            new Object[] { TransactionType.DEBIT, "Dog", -700000 },
+            new Object[] { TransactionType.DEBIT, "Dog toys", -1549 },
+            new Object[] { TransactionType.DEBIT, "Blue Audi", -4000000 },
+            new Object[] { TransactionType.DEBIT, "Small Yacht", -5900099 },
+            new Object[] { TransactionType.DEBIT, "Plastic Surgery", -2500000 },
+            new Object[] { TransactionType.DEBIT, "Skiing holiday", -510000 }
+
+    };
+
+    private Transaction[] seedTransactions() {
+        Transaction[] transactions = new Transaction[transactionData.length];
+        for (int i = 0; i < transactionData.length; i++) {
+            Object[] seedData = (Object[]) transactionData[i];
+            Transaction transaction = new Transaction();
+            transaction.setType((TransactionType) seedData[0]);
```

Comment:
Whenever we have to cast types like this (`(TransactionType) seedData[0]`) we've lost 'type safety' somewhere along the way - which is an important feature of the Java language which helps developers spot bugs earlier. 

For example: What happens if you accidentally change the Money (which is a `Long`) to a String? The compiler won't complain (as `String`s are `Object`s too) but when you run the application there will probably be a `ClassCastException`.

There are other advantages of ensuring we have type safety whenever possible as well, but this is by far the most important one.